### ISSUE TITLE
Some pytorch network weights have no gradient when --mixed-precision=True

### ISSUE THREAD

**Issue by yijicheng** [nerfstudio-project/nerfstudio#2434]
I add `print([[p.grad is None for p in g['params']] for g in optimizer.param_groups]) before`
https://github.com/nerfstudio-project/nerfstudio/blob/ef2fd3dbe1d5b6f35781716b747ea142fe21f5e9/nerfstudio/engine/optimizers.py#L130

And I run `ns-train instant-ngp --mixed-precision=False --pipeline.model.implementation=torch`, get `[[False, False, False, False, False, False, False, False, False, False, False, False]]`

run `ns-train instant-ngp --mixed-precision=True --pipeline.model.implementation=torch`, get `[[False, False, True, True, True, True, False, False, False, False, False, False]]`
